### PR TITLE
[Impeller] Document `impeller::Context`.

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

File: impeller/renderer/context.h [hunk 1]
```diff
@@ -19,31 +19,121 @@ class CommandBuffer;
 class PipelineLibrary;
 class Allocator;
 
+//------------------------------------------------------------------------------
+/// @brief      Do do anything rendering related with Impeller, you need a
```
Comment (jonahwilliams) [inline impeller/renderer/context.h:23]: nit: dodo

File: impeller/renderer/context.h [hunk 2]
```diff
@@ -19,31 +19,121 @@ class CommandBuffer;
...
+///             one in the process. The context represents a connection to a
+///             graphics or compute accelerator on the device.
+///
+///             If there are multiple context in a process, it would typically
+///             be for separation of concerns (say, use with multiple engines in
+///             Flutter), talking to multiple accelerators, or talking to the
+///             same accelerator using different client APIs (Metal, Vulkan,
+///             OpenGL ES, etc..).
+///
+///             Contexts are thread-safe. They may be created, used, and
```
Comment (jonahwilliams) [inline impeller/renderer/context.h:36]: There is a small caveat here in that they may still need to be destructed on the thread they were created from, at least as long as they own their own concurrent worker loop.

**Review by jonahwilliams — APPROVED** [review]
LGTM with nit

File: impeller/renderer/context.h (hunk 2 above)
Comment (jonahwilliams) [reply impeller/renderer/context.h:36]: actually, I guess not on the thread it was constructed on - just not on a worker thread from an internal pool

Comment (chinmaygarde) [reply impeller/renderer/context.h:36]: Added the clarification.

File: impeller/renderer/context.h (hunk 1 above)
Comment (chinmaygarde) [reply impeller/renderer/context.h:23]: Done,

**Pull request merged**